api/lock.go

```diff
@@ -123,23 +123,12 @@ func (l *Lock) Lock(stopCh <-chan struct{}) (<-chan struct{}, error) {
 	}
 
 	// Check if we need to create a session first
-	l.lockSession = l.opts.Session
```

Comment:
Not sure what you mean here. The SessionName and SessionTTL would still coming from the opts. It's the id created by createSession that would get updated. Maybe I'm not seeing a use case in which you would want that to happen outside of the Lock(), Unlock(), Destroy() API but it seems to me that it can all be handled internally.